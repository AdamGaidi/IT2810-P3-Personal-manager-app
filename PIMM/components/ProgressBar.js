import React, { Component } from "react";
import { ScrollView, StyleSheet, View, Text, Animated } from "react-native";
import Colors from "../constants/Colors";

export default class ProgressBar extends Component {
  componentWillMount() {
    this.animation = new Animated.Value(this.props.progress);
  }

  componentDidMount() {
    Animated.timing(this.animation, {
      toValue: this.props.progress,
      duration: this.props.duration
    }).start();
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevProps.progress !== this.props.progress) {
      Animated.timing(this.animation, {
        toValue: this.props.progress,
        duration: this.props.duration
      }).start();
    }
  }

  render() {
    const {
      height,
      borderColor,
      borderWidth,
      borderRadius,
      barColor,
      fillColor
    } = this.props;

    const widthInterpolated = this.animation.interpolate({
      inputRange: [0, 1],
      outputRange: ["0%", "100%"],
      extrapolate: "clamp"
    });

    const interpolateColor = this.animation.interpolate({
      inputRange: [0, 1],
      outputRange: ["#e55f00", "#b3e600"]
    });

    const animatedStyle = {
      backgroundColor: interpolateColor
    };

    return (
      <View style={[styles.progressBar, { height: height }]}>
        <View
          style={[
            styles.borderView,
            { borderColor, borderWidth, borderRadius }
          ]}
        >
          <View
            style={[
              StyleSheet.absoluteFill,
              { backgroundColor: fillColor, borderRadius }
            ]}
          />
          <Animated.View
            style={[
              styles.animation,
              animatedStyle,
              {
                width: widthInterpolated,
                borderRadius
              }
            ]}
          />
          <Text style={{ textAlign: "center", fontSize: 10 }}>
            {this.props.stepsToday} / {this.props.dailyGoal} steps
          </Text>
        </View>
      </View>
    );
  }
}

ProgressBar.defaultProps = {
  height: 18,
  borderColor: Colors.progressBarFill,
  borderWidth: 2,
  borderRadius: 10,
  fillColor: Colors.progressBarFill,
  duration: 100
};

const styles = StyleSheet.create({
  progressBar: {
    flex: 1,
    flexDirection: "row",
    margin: 5
  },
  borderView: {
    flex: 1
  },
  animation: {
    position: "absolute",
    left: 0,
    top: 0,
    bottom: 0
  }
});
