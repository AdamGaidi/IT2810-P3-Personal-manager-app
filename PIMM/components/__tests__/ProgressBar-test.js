import React from "react";
import ProgressBar from "../ProgressBar";
import ShallowRenderer from "react-test-renderer/shallow";

jest.mock("Animated", () => {
  return {
    createTimer: jest.fn(),
    timing: jest.fn(() => {
      return {
        start: jest.fn()
      };
    }),
    Value: jest.fn(() => {
      return {
        interpolate: jest.fn()
      };
    })
  };
});

it("renders correctly with example props", () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <ProgressBar progress={0.5} stepsToday={5000} dailyGoal={10000} />
  );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
