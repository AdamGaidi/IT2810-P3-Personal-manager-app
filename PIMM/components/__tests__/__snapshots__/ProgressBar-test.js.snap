// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with example props 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
        "margin": 5,
      },
      Object {
        "height": 18,
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "borderColor": "#bebebe",
          "borderRadius": 10,
          "borderWidth": 2,
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "#bebebe",
            "borderRadius": 10,
          },
        ]
      }
    />
    <UNDEFINED
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": undefined,
          },
          Object {
            "borderRadius": 10,
            "width": undefined,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#707070",
          "fontSize": 10,
          "textAlign": "center",
        }
      }
    >
      5000
       /
       
      10000
       steps
    </Text>
  </Component>
</Component>
`;
